### PR TITLE
Fix: Links to Docs page didn't work

### PR DESCRIPTION
**Pull request by reglim**
The docs page was way too complicated anyway, so I simplified it a bit.

I fixed the following issues:
* Link to docs page resulted in 404, when reloaded
* Browser Back button not working properly (Docs and Search)
* Open in New Tab not working 

fixes: #475, #476, #477, #479

### CODE REVIEW

File: web/src/components/Project.tsx [hunk 1]
```diff
@@ -12,12 +12,12 @@ interface Props {
   onFavoriteChanged: () => void
 }
 
-export default function Project(props: Props): JSX.Element {
+export default function Project (props: Props): JSX.Element {
```
Comment (randombenj) [inline web/src/components/Project.tsx:15]: Is this some js guideline?

File: web/src/pages/Docs.tsx [hunk 2]
```diff
@@ -163,19 +182,20 @@ export default function Docs(): JSX.Element {
...
+  if (versions.length === 0) {
+    return <LoadingPage />
+  }
+
   return (
     <>
       <iframe
         title="docs"
         ref={iFrameRef}
+        key={uniqueId()}
```
Comment (randombenj) [inline web/src/pages/Docs.tsx:198]: Why is this needed?

File: docat/docat/nginx/default [hunk 3]
```diff
@@ -23,5 +23,6 @@ server {
     }
 
     location / {
+        try_files $uri /index.html;
```
[the remark this answers is not in the document]
Comment (fliiiix) [reply docat/docat/nginx/default:26]: ideally we support all of that but i am also very willing to tell people they need a `index.html` to make it work

[the remark this answers is not in the document]
Comment (reglim) [reply docat/docat/nginx/default:26]: What this does is it redirects all requests to `/...` to the index.html file, which is the output of react build. This just makes sure that it falls back to the page, otherwise we would get an NGINX 404

**Comment by reglim** [comment]
> i had a quick look no idea what that does but seems fine to me if it works

Here's an explanation: https://stackoverflow.com/a/17800131/20919151

File: docat/docat/nginx/default (hunk 3 above)
[the remark this answers is not in the document]
Comment (randombenj) [reply docat/docat/nginx/default:26]: Ah, this is only for our internal ui, then we can leave it as is, sorry for the misunderstanding.

**Comment by reglim** [comment]
> > > i had a quick look no idea what that does but seems fine to me if it works
> > 
> > 
> > Here's an explanation: https://stackoverflow.com/a/17800131/20919151
> 
> This is something I usually try to add as a source code comment so it is more clear when reading the code ;)

Fair, I just thought it's such a common pattern in nginx configs that I didn't bother ...

File: web/src/pages/Docs.tsx [hunk 4]
```diff
@@ -4,189 +4,168 @@
...
-
-      setVersion(latestVersion)
-      updateRoute(project, latestVersion, page, hideUi)
-    } catch (e) {
-      console.error(e)
-      setLoadingFailed(true)
+  const onIFrameLocationChanged = (url: string): void => {
+    url = url.split('/doc/')[1]
+    if (url.length === 0) {
+      // should never happen
```
Comment (fliiiix) [inline web/src/pages/Docs.tsx:74]: log an error here?

Comment (reglim) [reply web/src/pages/Docs.tsx:75]: I mean we could, but I didn't think it would be interesting

**Review by randombenj — APPROVED** [review]
